### PR TITLE
[Service] Rename the `ServiceId` proto to `Service`

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

Renamed the `ServiceId` proto to `Service` so `ServiceId.Id` is more semantic and less confusing.

## Issue

- Thought it is not directly related to #140, it was driven out of confusion by the author while beginning to implement this piece of work

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: pkg/relayer/proxy/interface.go [hunk 1]
```diff
@@ -41,6 +41,6 @@ type RelayServer interface {
 	// Stop terminates the service server and returns an error if it fails.
 	Stop(ctx context.Context) error
 
-	// ServiceId returns the serviceId of the service.
-	ServiceId() *sharedtypes.ServiceId
+	// Service returns the serviceId of the service.
```
Comment (bryanchriswhite) [inline pkg/relayer/proxy/interface.go:44]:
```suggestion
	// Service returns the service to which the RelayServer relays.
```

File: pkg/relayer/proxy/jsonrpc.go [hunk 2]
```diff
@@ -12,8 +12,8 @@ import (
 var _ RelayServer = (*jsonRPCServer)(nil)
 
 type jsonRPCServer struct {
-	// serviceId is the id of the service that the server is responsible for.
-	serviceId *sharedtypes.ServiceId
+	// service is the id of the service that the server is responsible for.
```
Comment (bryanchriswhite) [inline pkg/relayer/proxy/jsonrpc.go:15]:
```suggestion
	// service is the service that the server is responsible for.
```

File: pkg/relayer/proxy/jsonrpc.go [hunk 3]
```diff
@@ -71,9 +71,9 @@ func (j *jsonRPCServer) Stop(ctx context.Context) error {
 	return j.server.Shutdown(ctx)
 }
 
-// ServiceId returns the serviceId of the JSON-RPC service.
-func (j *jsonRPCServer) ServiceId() *sharedtypes.ServiceId {
-	return j.serviceId
+// Service returns the serviceId of the JSON-RPC service.
```
Comment (bryanchriswhite) [inline pkg/relayer/proxy/jsonrpc.go:74]:
```suggestion
// Service returns the JSON-RPC service.
```

File: proto/pocket/shared/service.proto [hunk 4]
```diff
@@ -6,24 +6,19 @@ package pocket.shared;
...
 option go_package = "github.com/pokt-network/poktroll/x/shared/types";
 
-// TODO_CLEANUP(@Olshansk): Add native optional identifiers once its supported; https://github.com/ignite/cli/issues/3698
-
-// ServiceId message to encapsulate unique and semantic identifiers for a service on the network
-message ServiceId {
-    // NOTE: `ServiceId.Id` may seem redundant but was desigtned created to enable more complex service identification
+// Service message to encapsulate unique and semantic identifiers for a service on the network
+message Service {
+    // NOTE: `Service.Id` may seem redundant but was desigtned created to enable more complex service identification
```
Comment (bryanchriswhite) [inline proto/pocket/shared/service.proto:11]: I think this comment may now be irrelevant. Wdyt?

```suggestion
```

File: proto/pocket/shared/service.proto [hunk 5]
```diff
@@ -6,24 +6,19 @@ package pocket.shared;
 
 option go_package = "github.com/pokt-network/poktroll/x/shared/types";
 
-// TODO_CLEANUP(@Olshansk): Add native optional identifiers once its supported; https://github.com/ignite/cli/issues/3698
```
Comment (bryanchriswhite) [inline proto/pocket/shared/service.proto:9]: Did this get resolved?

File: proto/pocket/shared/service.proto [hunk 6]
```diff
@@ -6,24 +6,19 @@ package pocket.shared;
...
     string name = 2; // (Optional) Semantic human readable name for the service
-
-    // NOTE: `ServiceId.Id` may seem redundant but was designed to enable more complex service identification.
-    // For example, what if we want to request a session for a certain service but with some additional configs that identify it?
 }
 
 // ApplicationServiceConfig holds the service configuration the application stakes for
 message ApplicationServiceConfig {
-    ServiceId service_id = 1; // Unique and semantic identifier for the service
+    Service service = 1; // Unique and semantic identifier for the service
```
Comment (bryanchriswhite) [inline proto/pocket/shared/service.proto:21]: This comment is now outdated.

File: x/shared/helpers/service.go [hunk 7]
```diff
@@ -25,6 +27,11 @@ func init() {
 
 }
 
+// IsValidService checks if the input service is valid
+func IsValidService(service *sharedtypes.Service) bool {
+	return service != nil && IsValidServiceId(service.Id) && IsValidServiceName(service.Name)
+}
+
```
Comment (bryanchriswhite) [inline x/shared/helpers/service.go:34]: Nice refactor! :ok_hand:

File: x/shared/helpers/service_test.go [hunk 8]
```diff
@@ -1,78 +1,181 @@
...
+			expected: false,
+		},
+		{
+			testCase: "Name exceeds max length",
+			id:       "ValidID",
+			name:     "This service name is way too long to be considered valid since it exceeds the max length",
+			expected: false,
+		},
+		{
+			testCase: "Empty ID",
```
Comment (bryanchriswhite) [inline x/shared/helpers/service_test.go:41]: Wdyt about explicitly stating the expectation in the subtest name, like the others do:

```suggestion
			testCase: "Empty ID is invalid",
```

File: x/shared/helpers/service_test.go [hunk 9]
```diff
@@ -1,78 +1,181 @@
...
+		t.Run(test.testCase, func(t *testing.T) {
+			service := &sharedtypes.Service{
+				Id:   test.id,
+				Name: test.name,
+			}
+			result := IsValidService(service)
+			if result != test.expected {
+				t.Errorf("Test Case '%s' - IsValidService() with Id: '%s', Name: '%s', expected %v, got %v",
+					test.testCase, test.id, test.name, test.expected, result)
+			}
```
Comment (bryanchriswhite) [inline x/shared/helpers/service_test.go:64]: Why not use `require`?

```suggestion
			require.Equal(t, test.expected, result)
```

File: x/shared/helpers/service_test.go [hunk 10]
```diff
@@ -1,78 +1,181 @@
...
+		})
+	}
+}
 
 func TestIsValidServiceId(t *testing.T) {
 	tests := []struct {
+		testCase string
```
Comment (bryanchriswhite) [inline x/shared/helpers/service_test.go:71]: NIT: wdyt about `name` instead?

```suggestion
		name string
```

File: x/supplier/keeper/supplier.go [hunk 11]
```diff
@@ -64,5 +64,5 @@ func (k Keeper) GetAllSupplier(ctx sdk.Context) (suppliers []sharedtypes.Supplie
 	return
 }
 
-// TODO_OPTIMIZE: Index suppliers by serviceId so we can easily query `k.GetAllSupplier(ctx, ServiceId)`
+// TODO_OPTIMIZE: Index suppliers by serviceId so we can easily query `k.GetAllSupplier(ctx, Service)`
```
Comment (bryanchriswhite) [inline x/supplier/keeper/supplier.go:67]:
```suggestion
// TODO_OPTIMIZE: Index suppliers by service so we can easily query `k.GetAllSupplier(ctx, Service)`
```

File: x/shared/helpers/service_test.go (hunk 10 above)
Comment (bryanchriswhite) [reply x/shared/helpers/service_test.go:71]: I'm noticing that ignite generated tests seem to use the word `desc`:

image

File: proto/pocket/shared/service.proto [hunk 12]
```diff
@@ -8,22 +8,18 @@ option go_package = "github.com/pokt-network/poktroll/x/shared/types";
...
     string name = 2; // (Optional) Semantic human readable name for the service
-
-    // NOTE: `ServiceId.Id` may seem redundant but was designed to enable more complex service identification.
-    // For example, what if we want to request a session for a certain service but with some additional configs that identify it?
 }
 
 // ApplicationServiceConfig holds the service configuration the application stakes for
 message ApplicationServiceConfig {
-    ServiceId service_id = 1; // Unique and semantic identifier for the service
+    Service service = 1; // The Service for which the application is configured for
```
Comment (bryanchriswhite) [inline proto/pocket/shared/service.proto:22]: NIT: I think it reads slightly better if we  drop  the "for" at the end:

```suggestion
    Service service = 1; // The Service for which the application is configured
```

File: proto/pocket/shared/service.proto [hunk 13]
```diff
@@ -32,7 +28,7 @@ message ApplicationServiceConfig {
 
 // SupplierServiceConfig holds the service configuration the supplier stakes for
 message SupplierServiceConfig {
-    ServiceId service_id = 1; // Unique and semantic identifier for the service
+    Service service = 1; // The Service for which the supplier is configured for
```
Comment (bryanchriswhite) [inline proto/pocket/shared/service.proto:31]:
```suggestion
    Service service = 1; // The Service for which the supplier is configured
```

**Review by bryanchriswhite — APPROVED** [review]
Looking good! :raised_hands::ship:

File: proto/pocket/shared/service.proto (hunk 12 above)
Comment (Olshansk) [reply proto/pocket/shared/service.proto:22]: Didn't notice this so will tend to it in #123

File: proto/pocket/shared/service.proto (hunk 13 above)
Comment (Olshansk) [reply proto/pocket/shared/service.proto:31]: Didn't notice this so will tend to it in #123

**Pull request merged**